File: context_os/pipeline/middlewares/optimize_middleware.py
"""OptimizeMiddleware — 上下文优化。"""

from context_os.optimizer.optimizer import ContextOptimizer
from context_os.pipeline.context import PipelineContext
from context_os.pipeline.middleware import PipelineMiddleware


class OptimizeMiddleware(PipelineMiddleware):
    """排序记忆 → 压缩对话 → 分配预算。"""

    def __init__(self, optimizer: ContextOptimizer):
        self._o = optimizer

    def name(self) -> str:
        return "optimize"

    def order(self) -> int:
        return 400

    async def execute(self, ctx: PipelineContext) -> None:
        ctx.optimized_context = await self._o.optimize(
            ctx.unified_context, ctx.task_spec
        )

File: java-context-os/src/main/java/com/owencli/contextos/pipeline/middleware/OptimizeMiddleware.java
package com.owencli.contextos.pipeline.middleware;

import com.owencli.contextos.optimizer.ContextOptimizer;
import com.owencli.contextos.pipeline.PipelineContext;
import com.owencli.contextos.pipeline.PipelineMiddleware;
import java.util.concurrent.CompletableFuture;

public class OptimizeMiddleware implements PipelineMiddleware {
    private final ContextOptimizer o;
    public OptimizeMiddleware(ContextOptimizer co) { this.o = co; }
    public String name() { return "optimize"; }
    public int order() { return 400; }
    public CompletableFuture<Void> execute(PipelineContext ctx) {
        return o.optimize(ctx.unifiedContext(), ctx.taskSpec()).thenAccept(ctx::setOptimizedContext);
    }
}
